libpromises/rlist.c

```diff
@@ -1206,35 +1204,35 @@ void RlistWrite(Writer *writer, const Rlist *list)
     WriterWriteChar(writer, '}');
 }
 
-/* Note: only single quotes are escaped, as they are used in RlistWrite to
-   delimit strings. If double quotes would be escaped, they would be mangled by
-   RlistParseShown */
-
-static void ScalarWrite(Writer *w, const char *s)
+void ScalarWrite(Writer *writer, const char *s, bool dontQuote)
```

Comment:
You can leave this one with the extra bool, since it is internal to this file. However, I ask that you flip the meaning of the flag, IOW use `bool quote` instead of `bool dontQuote`, so you avoid the confusing double negative. This means you'll have to flip the true/false flags where you call it as well, of course.